### DISCUSSION TITLE
What are the basics of using Drupal.announce() in practice?

### DISCUSSION THREAD

**Discussion by cfbauer** [CivicActions/accessibility#690] [Q&A]
Here's the [documentation on Drupal.org](https://www.drupal.org/docs/drupal-apis/javascript-api/accessibility-tools-for-javascript-in-drupal-8):

> Many page updates are expressed visually through color changes and animations. In order to make page updates apparent in a non-visual way, Drupal provides the Drupal.announce JavaScript method. This method creates an [aria-live](https://www.w3.org/TR/wai-aria-1.1/#aria-live) element on the page. Text appended to this node is read by a screen-reading user agent.
> 
> Drupal.announce accepts a string to be read by an audio UA. You can also set a second parameter: the priority. Here are a couple examples:

```Drupal.announce('Entering edit mode');
Drupal.announce(
  Drupal.t('You look beautiful today.')
);
Drupal.announce(
  Drupal.t('Please fill in your user name'),
  'assertive'
);
```

I'd love to see some real world examples!

**Comment by dmundra** [comment]
Found one post on the `Drupal.announce()` https://thinkshout.com/blog/2020/11/screen-readers/ that provided an example with using and how it works without using it.

**Comment by mgifford** [comment]
This posed an interesting discussion @cfbauer. Thanks for asking it here. 

My first reaction was to want to `grep -r 'Drupal.announce('` and see what interesting stuff I could pull out. Also looking at the [Drupal.org issue queue](https://www.drupal.org/project/issues/search?text=announce&projects=&assigned=&submitted=&project_issue_followers=&status%5B%5D=Open&issue_tags_op=%3D&issue_tags=Accessibility). 

But would be good to have some more direction for sure. Maybe we should write some clear instructions for Drupal 10? 

@dmundra thanks for the ThinkShout article.
